### PR TITLE
Issue 114 wgs spark coverage

### PR DESCRIPTION
**Pull request by achevali**
Add WGS spark coverage
Automatically create and emit wgs bedfile
Update to late pre_alpha gatk public

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import java.util.stream.Collectors;
+
+@CommandLineProgramProperties(
+        summary = "Spark WGS Coverage",
+        oneLineSummary = "Spark WGS Coverage.",
+        programGroup = CopyNumberProgramGroup.class)
+public class SparkGenomeReadCounts extends GATKSparkTool {
+    private static final long serialVersionUID = 1l;
+
+    private static final int binsize = 10000;
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java]: Add an issue to allow binsize specification through the CLI.  ``--binsize 1000``

File: src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+    protected static final String OUTPUT_FILE_SHORT_NAME = "o";
+    protected static final String OUTPUT_FILE_LONG_NAME = "outputFile";
+
+    @Argument(doc = "Output tsv file for the SVD U.  No headers, just tab-separated values",
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java]: Copy and paste error?

File: src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        byKeyMutable.putAll(byKey);
+        missingGenomeBins.stream().forEach(m -> byKeyMutable.put(m, 0l));
+
+        final SortedMap<SimpleInterval, Long> byKeySorted = new TreeMap<>(IntervalUtils.LEXICOGRAPHICAL_ORDER_COMPARATOR);
+        byKeySorted.putAll(byKeyMutable);
+
+        TargetCoverageUtils.writeTargetsWithCoverageFromSimpleInterval(outputFile, sampleName, byKeySorted, comments);
+    }
+
+    public static List<SimpleInterval> createFullGenomeBins(final int binsize, final SAMSequenceDictionary sequenceDictionary){
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java]: public method needs javadoc

File: src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        final List<SimpleInterval> sequenceIntervals = IntervalUtils.getAllIntervalsForReference(sequenceDictionary);
+        return IntervalUtils.cutToShards(sequenceIntervals, binsize);
+    }
+
+    public static SimpleInterval createKey(final GATKRead read, final SAMSequenceDictionary sequenceDictionary){
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java]: public method needs javadoc

File: src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+        return new SimpleInterval(contig, newStart, newEnd);
+    }
+
+    @Override
+    protected void runTool(JavaSparkContext ctx) {
+        collectReads(ctx);
+    }
+
+    private void print(SortedMap<SimpleInterval, Long> byKeySorted, PrintStream ps) {
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java]: Please delete

File: src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        ps.println("CONTIG" + "\t" + "START" + "\t" + "END" + "\t" + "<ALL>");
+        for (final Locatable loc : byKeySorted.keySet()){
+            ps.println(loc.getContig() + "\t" + loc.getStart() + "\t" + loc.getEnd() + "\t" + byKeySorted.get(loc));
+        }
+    }
+
+    public ReadFilter makeReadFilter() {
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java]: public method needs javadoc or make it private.  I suggest making it private.

File: src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+        return new WellformedReadFilter(getHeaderForReads())
+                .and(ReadFilterLibrary.MAPPED)
+                .and(ReadFilterLibrary.NOT_DUPLICATE)
+                .and(ReadFilterLibrary.NON_ZERO_REFERENCE_LENGTH_ALIGNMENT)
+                .and(ReadFilterLibrary.PASSES_VENDOR_QUALITY_CHECK);
+    }
+
+    public List<Target> createTargetCollectionFromSimpleInterval(final List<SimpleInterval> intervalList){
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java]: Again.... public method needs javadoc or make it private. I suggest making it private.

**Comment by LeeTL1220** [comment]
Please move the testdata into ``org.broadinstitute.hellbender.tools.exome`` 

NOT: ``org.broadinstitute.tools.exome``

**Comment by LeeTL1220** [comment]
Please change all unit/integration tests from ``--VERBOSITY`` to ``--verbosity`` to make them pass.

File: src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java (hunk 1 above)
Comment (achevali) [inline src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java]: Issue added

File: src/main/java/org/broadinstitute/hellbender/tools/exome/TargetCoverageUtils.java [hunk 8]
```diff
@@ -115,6 +117,62 @@ public static void writeTargetsWithCoverage(final File outFile, final String sam
     }
 
     /**
+     * write a list of targets with coverage to file with dummy names
+     *
+     */
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/TargetCoverageUtils.java]: Needs parameters in javadoc

**Pull request merged**